### PR TITLE
feat: add date picker for journal [wip]

### LINKED ISSUE #2717: Date Picker for Journal
I'd like to be able to find out what a did on a certain day. If there was a date picker in the journal, I could see all diary and activity entries from that date.

### PR DESCRIPTION
**Pull request by Jordanjun98**
This Fix #2717

#### Before submitting the PR
- [x] Read the [CONTRIBUTING document](https://github.com/monicahq/monica/blob/master/CONTRIBUTING.md) before submitting your PR.
- [x] If the PR is related to an issue or fix one, don't forget to indicate it.
- [x] Indicate `[wip]` in the title of the PR it is is not final yet. Remove `[wip]` when ready. Otherwise the PR will be considered complete and rejected if it's not working.

### General checks
- [x] Make sure that the change you propose is the smallest possible.
- [x] The name of the PR should follow the [conventional commits guideline](https://github.com/monicahq/monica/blob/master/docs/contribute/index.md#conventional-commits) that the project follows.

### Front-end changes
- [ ] If you change the UI, make sure to ask repositories administrators first about your changes by pinging @djaiss or @asbiin in this PR.
- [ ] Screenshots are included if the PR changes the UI.
- [ ] Front-end tests have been written with Cypress.

#### Backend/models changes
- [ ] The API has been updated.
- [ ] API's documentation has been added by submitting a pull request in the [marketing website repository](https://github.com/monicahq/marketing_site/pulls).
- [ ] Tests have been added for the new code.
- [ ] If you change a model, make sure the FakeContentTableSeeder is updated. We need seeders to develop locally and generate fake data.

#### If the code changes the SQL schema
- [x] Make sure exporting account data as SQL is still working.
- [x] Make sure your changes do not break importing data with `vCard` and `.csv` files.
- [x] Make sure account reset and deletion still work.

#### Other tasks
- [ ] [CHANGELOG](https://github.com/monicahq/monica/blob/master/CHANGELOG) entry added, if necessary, under `UNRELEASED`.
- [ ] [CONTRIBUTORS](https://github.com/monicahq/monica/blob/master/CONTRIBUTORS) entry added, if necessary.
- [ ] If it's relevant and worth mentioning, create a changelog entry for this change. The changelog entry will appear inside the UI for all users to see. To know if your change is worth the creation of a changelog entry, [read the documentation](https://github.com/monicahq/monica/blob/master/docs/administrators/tips.md#when-is-it-relevant-to-create-a-changelog-entry).
- [ ] Don't forget to [ask for a free account](mailto:regis@monicahq.com) on https://monicahq.com as anyone who contributes can request a free account.

### CODE REVIEW

**Comment by tbirrell** [comment]
You have a lot of extra files here that shouldn't be part of the PR.
```
.env.dev 
composer.lock 
package-lock.json
public/css/app-ltr.css 
public/css/app-rtl.css 
public/css/stripe.css 
public/js/app.js 
public/js/manifest.js 
public/js/stripe.js 
public/js/u2f-api.js 
public/js/vendor.js 
public/mix-manifest.json 
```
Most of these are files that get changed by virtue of running `yarn run dev`, so to include them is pointless since anyone's build process will just overwrite them.

This only files you need for the PR are:
```
app/Http/Controllers/JournalController.php 
resources/assets/js/components/journal/JournalList.vue
package.json
```

To correct this, you can selectively pull in the files from master so that they no longer have a diff in the PR. That process would look something like this.

```
git checkout master
git pull monicahq master
git checkout feature/datePickerJournal
git checkout master -- path/filename.php (repeating this step as necessary)
```

See this [StackOverflow Q/A](https://stackoverflow.com/questions/2364147/how-to-get-just-one-file-from-another-branch/2364223) to see more details.

File: .env.dev [hunk 1]
```diff
@@ -11,8 +11,8 @@ DB_CONNECTION=mysql
 DB_HOST=mysql
 DB_PORT=3306
 DB_DATABASE=monica
-DB_USERNAME=homestead
-DB_PASSWORD=secret
+DB_USERNAME=root
```
Comment (asbiin) [inline .env.dev]: Please don't change this
```suggestion
DB_USERNAME=homestead
```

File: .env.dev [hunk 2]
```diff
@@ -11,8 +11,8 @@ DB_CONNECTION=mysql
...
+DB_PASSWORD=
```
Comment (asbiin) [inline .env.dev]:
```suggestion
DB_PASSWORD=secret
```

File: .gitignore [hunk 3]
```diff
@@ -3,9 +3,11 @@
 /node_modules
 /public/storage
 /persist
+/public
```
Comment (asbiin) [inline .gitignore]: We need this directory ...
```suggestion
```

File: .gitignore [hunk 4]
```diff
@@ -3,9 +3,11 @@
...
 Homestead.yaml
 Homestead.json
 .env
+.env.dev
```
Comment (asbiin) [inline .gitignore]: Not sure how to use it
```suggestion
```

File: resources/assets/js/components/journal/JournalList.vue [hunk 5]
```diff
@@ -12,6 +12,8 @@
       <!-- How was your day -->
       <journal-rate-day @hasRated="hasRated" />
 
+      <datepicker placeholder="Select Date"  @selected="getEntriesDate" v-model="date" ></datepicker>
```
Comment (asbiin) [inline resources/assets/js/components/journal/JournalList.vue]: You should use `<form-date>` component here, which use `datepicker`

File: package.json [hunk 6]
```diff
@@ -64,7 +64,8 @@
     "vue-i18n": "^8.0.0",
     "vue-js-toggle-button": "^1.2.2",
     "vue-notification": "^1.3.6",
-    "vue-select": "^3.0.2"
+    "vue-select": "^3.0.2",
+    "vuejs-datepicker": "^1.5.4"
```
Comment (asbiin) [inline package.json]: Please don't add ``vuejs-datepicker``, we already use ``@hokify/vuejs-datepicker``.
You can use ``<form-date>`` component which use it

**Comment by asbiin** [comment]
I fixed the title of you PR, adding ``[wip]`` tag. Be sure to remove it when it's finished.

**Comment by asbiin** [comment]
Closing this after long delay without answer

**Pull request closed without merging**